<?xml version="1.0" encoding="UTF-8"?>
<testng-results skipped="0" failed="3" total="7" passed="4">
  <reporter-output>
    <line>
      <![CDATA[用例ID: 1]]>
    </line>
    <line>
      <![CDATA[用例名称: 使用正确的账号，正确的密码，可以正常登录成功，且接口返回的实际结果都正确]]>
    </line>
    <line>
      <![CDATA[请求地址: http://120.79.232.23:8000/api/user/login]]>
    </line>
    <line>
      <![CDATA[请求方式: POST]]>
    </line>
    <line>
      <![CDATA[请求参数: username=admin&password=admin369874125]]>
    </line>
    <line>
      <![CDATA[返回结果: {"code":"999999","msg":"成功","data":{"first_name":"admin","last_name":"","phone":"12345678911","email":"admin@163.com","key":"640ca3dbb875b6cfe898bf0c9b0f45a921ef169b","date_joined":"2018-06-28 02:56:00","userphoto":"/file/userphoto.jpg"}}]]>
    </line>
    <line>
      <![CDATA[用例结果: 〖first_name〗=>Expected: 【admin】, Actual: 【admin】]]>
    </line>
    <line>
      <![CDATA[用例结果: 〖last_name〗=>Expected: 【】, Actual: 【】]]>
    </line>
    <line>
      <![CDATA[用例结果: 〖phone〗=>Expected: 【12345678911】, Actual: 【12345678911】]]>
    </line>
    <line>
      <![CDATA[用例结果: 〖email〗=>Expected: 【admin@163.com】, Actual: 【admin@163.com】]]>
    </line>
    <line>
      <![CDATA[用例结果: 〖key〗=>Expected: 【640ca3dbb875b6cfe898bf0c9b0f45a921ef169b】, Actual: 【640ca3dbb875b6cfe898bf0c9b0f45a921ef169b】]]>
    </line>
    <line>
      <![CDATA[用例结果: 〖date_joined〗=>Expected: 【2018-06-28 02:56:00】, Actual: 【2018-06-28 02:56:00】]]>
    </line>
    <line>
      <![CDATA[用例结果: 〖userphoto〗=>Expected: 【/file/userphoto.jpg】, Actual: 【/file/userphoto.jpg】]]>
    </line>
    <line>
      <![CDATA[用例ID: 2]]>
    </line>
    <line>
      <![CDATA[用例名称: 使用空值的账号，空值的密码，无法正常登录成功，且接口返回的实际结果都正确]]>
    </line>
    <line>
      <![CDATA[请求地址: http://120.79.232.23:8000/api/user/login]]>
    </line>
    <line>
      <![CDATA[请求方式: POST]]>
    </line>
    <line>
      <![CDATA[请求参数: username=&password=]]>
    </line>
    <line>
      <![CDATA[返回结果: {"username":["该字段不能为空。"],"password":["该字段不能为空。"],"code":400}]]>
    </line>
    <line>
      <![CDATA[用例ID: 3]]>
    </line>
    <line>
      <![CDATA[用例名称: 使用空值的账号，正确的密码，无法正常登录成功，且接口返回的实际结果都正确]]>
    </line>
    <line>
      <![CDATA[请求地址: http://120.79.232.23:8000/api/user/login]]>
    </line>
    <line>
      <![CDATA[请求方式: POST]]>
    </line>
    <line>
      <![CDATA[请求参数: username=&password=admin369874125]]>
    </line>
    <line>
      <![CDATA[返回结果: {"username":["该字段不能为空。"],"code":400}]]>
    </line>
    <line>
      <![CDATA[用例ID: 4]]>
    </line>
    <line>
      <![CDATA[用例名称: 使用正确的账号，空值的密码，无法正常登录成功，且接口返回的实际结果都正确]]>
    </line>
    <line>
      <![CDATA[请求地址: http://120.79.232.23:8000/api/user/login]]>
    </line>
    <line>
      <![CDATA[请求方式: POST]]>
    </line>
    <line>
      <![CDATA[请求参数: username=admin&password=]]>
    </line>
    <line>
      <![CDATA[返回结果: {"password":["该字段不能为空。"],"code":400}]]>
    </line>
    <line>
      <![CDATA[用例ID: 5]]>
    </line>
    <line>
      <![CDATA[用例名称: 使用错误的账号，错误的密码，无法正常登录成功，且接口返回的实际结果都正确]]>
    </line>
    <line>
      <![CDATA[请求地址: http://120.79.232.23:8000/api/user/login]]>
    </line>
    <line>
      <![CDATA[请求方式: POST]]>
    </line>
    <line>
      <![CDATA[请求参数: username=123456&password=123456789]]>
    </line>
    <line>
      <![CDATA[返回结果: {"code":"999984","msg":"账号或密码错误"}]]>
    </line>
    <line>
      <![CDATA[用例结果: 〖msg〗=>Expected: 【账号或密码错误】, Actual: 【账号或密码错误】]]>
    </line>
    <line>
      <![CDATA[用例ID: 6]]>
    </line>
    <line>
      <![CDATA[用例名称: 使用错误的账号，正确的密码，无法正常登录成功，且接口返回的实际结果都正确]]>
    </line>
    <line>
      <![CDATA[请求地址: http://120.79.232.23:8000/api/user/login]]>
    </line>
    <line>
      <![CDATA[请求方式: POST]]>
    </line>
    <line>
      <![CDATA[请求参数: username=123456&password=admin369874125]]>
    </line>
    <line>
      <![CDATA[返回结果: {"code":"999984","msg":"账号或密码错误"}]]>
    </line>
    <line>
      <![CDATA[用例结果: 〖msg〗=>Expected: 【账号或密码错误】, Actual: 【账号或密码错误】]]>
    </line>
    <line>
      <![CDATA[用例ID: 7]]>
    </line>
    <line>
      <![CDATA[用例名称: 使用正确的账号，错误的密码，无法正常登录成功，且接口返回的实际结果都正确]]>
    </line>
    <line>
      <![CDATA[请求地址: http://120.79.232.23:8000/api/user/login]]>
    </line>
    <line>
      <![CDATA[请求方式: POST]]>
    </line>
    <line>
      <![CDATA[请求参数: username=admin&password=123456789]]>
    </line>
    <line>
      <![CDATA[返回结果: {"code":"999984","msg":"账号或密码错误"}]]>
    </line>
    <line>
      <![CDATA[用例结果: 〖msg〗=>Expected: 【账号或密码错误】, Actual: 【账号或密码错误】]]>
    </line>
  </reporter-output>
  <suite name="Suite" duration-ms="0" started-at="2018-10-30T10:02:52Z" finished-at="2018-10-30T10:02:52Z">
    <groups>
    </groups>
  </suite> <!-- Suite -->
  <suite name="Web后端-接口自动化测试" duration-ms="30446" started-at="2018-10-30T10:02:21Z" finished-at="2018-10-30T10:02:52Z">
    <groups>
    </groups>
    <test name="用户登录接口" duration-ms="30446" started-at="2018-10-30T10:02:21Z" finished-at="2018-10-30T10:02:52Z">
      <class name="TestCases.Login">
        <test-method status="PASS" signature="Stup()[pri:0, instance:TestCases.Login@39ba5a14]" name="Stup" is-config="true" duration-ms="15181" started-at="2018-10-30T18:02:21Z" finished-at="2018-10-30T18:02:36Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- Stup -->
        <test-method status="PASS" signature="Case1()[pri:0, instance:TestCases.Login@39ba5a14]" name="Case1" duration-ms="4342" started-at="2018-10-30T18:02:36Z" finished-at="2018-10-30T18:02:41Z">
          <reporter-output>
            <line>
              <![CDATA[用例ID: 1]]>
            </line>
            <line>
              <![CDATA[用例名称: 使用正确的账号，正确的密码，可以正常登录成功，且接口返回的实际结果都正确]]>
            </line>
            <line>
              <![CDATA[请求地址: http://120.79.232.23:8000/api/user/login]]>
            </line>
            <line>
              <![CDATA[请求方式: POST]]>
            </line>
            <line>
              <![CDATA[请求参数: username=admin&password=admin369874125]]>
            </line>
            <line>
              <![CDATA[返回结果: {"code":"999999","msg":"成功","data":{"first_name":"admin","last_name":"","phone":"12345678911","email":"admin@163.com","key":"640ca3dbb875b6cfe898bf0c9b0f45a921ef169b","date_joined":"2018-06-28 02:56:00","userphoto":"/file/userphoto.jpg"}}]]>
            </line>
            <line>
              <![CDATA[用例结果: 〖first_name〗=>Expected: 【admin】, Actual: 【admin】]]>
            </line>
            <line>
              <![CDATA[用例结果: 〖last_name〗=>Expected: 【】, Actual: 【】]]>
            </line>
            <line>
              <![CDATA[用例结果: 〖phone〗=>Expected: 【12345678911】, Actual: 【12345678911】]]>
            </line>
            <line>
              <![CDATA[用例结果: 〖email〗=>Expected: 【admin@163.com】, Actual: 【admin@163.com】]]>
            </line>
            <line>
              <![CDATA[用例结果: 〖key〗=>Expected: 【640ca3dbb875b6cfe898bf0c9b0f45a921ef169b】, Actual: 【640ca3dbb875b6cfe898bf0c9b0f45a921ef169b】]]>
            </line>
            <line>
              <![CDATA[用例结果: 〖date_joined〗=>Expected: 【2018-06-28 02:56:00】, Actual: 【2018-06-28 02:56:00】]]>
            </line>
            <line>
              <![CDATA[用例结果: 〖userphoto〗=>Expected: 【/file/userphoto.jpg】, Actual: 【/file/userphoto.jpg】]]>
            </line>
          </reporter-output>
        </test-method> <!-- Case1 -->
        <test-method status="FAIL" signature="Case2()[pri:0, instance:TestCases.Login@39ba5a14]" name="Case2" duration-ms="1571" started-at="2018-10-30T18:02:41Z" finished-at="2018-10-30T18:02:42Z">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[msg =>Expected 【账号和密码不能为空】 but found 【null】]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: msg =>Expected 【账号和密码不能为空】 but found 【null】
	at org.testng.Assert.fail(Assert.java:94)
	at com.jmoney.luckeylink.handler.Login_Handler.checkEquals(Login_Handler.java:312)
	at com.jmoney.luckeylink.handler.Login_Handler.Login(Login_Handler.java:288)
	at TestCases.Login.Case2(Login.java:23)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:86)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
	at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
            <line>
              <![CDATA[用例ID: 2]]>
            </line>
            <line>
              <![CDATA[用例名称: 使用空值的账号，空值的密码，无法正常登录成功，且接口返回的实际结果都正确]]>
            </line>
            <line>
              <![CDATA[请求地址: http://120.79.232.23:8000/api/user/login]]>
            </line>
            <line>
              <![CDATA[请求方式: POST]]>
            </line>
            <line>
              <![CDATA[请求参数: username=&password=]]>
            </line>
            <line>
              <![CDATA[返回结果: {"username":["该字段不能为空。"],"password":["该字段不能为空。"],"code":400}]]>
            </line>
          </reporter-output>
        </test-method> <!-- Case2 -->
        <test-method status="FAIL" signature="Case3()[pri:0, instance:TestCases.Login@39ba5a14]" name="Case3" duration-ms="1478" started-at="2018-10-30T18:02:42Z" finished-at="2018-10-30T18:02:44Z">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[msg =>Expected 【账号不能为空】 but found 【null】]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: msg =>Expected 【账号不能为空】 but found 【null】
	at org.testng.Assert.fail(Assert.java:94)
	at com.jmoney.luckeylink.handler.Login_Handler.checkEquals(Login_Handler.java:312)
	at com.jmoney.luckeylink.handler.Login_Handler.Login(Login_Handler.java:288)
	at TestCases.Login.Case3(Login.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:86)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
	at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
            <line>
              <![CDATA[用例ID: 3]]>
            </line>
            <line>
              <![CDATA[用例名称: 使用空值的账号，正确的密码，无法正常登录成功，且接口返回的实际结果都正确]]>
            </line>
            <line>
              <![CDATA[请求地址: http://120.79.232.23:8000/api/user/login]]>
            </line>
            <line>
              <![CDATA[请求方式: POST]]>
            </line>
            <line>
              <![CDATA[请求参数: username=&password=admin369874125]]>
            </line>
            <line>
              <![CDATA[返回结果: {"username":["该字段不能为空。"],"code":400}]]>
            </line>
          </reporter-output>
        </test-method> <!-- Case3 -->
        <test-method status="FAIL" signature="Case4()[pri:0, instance:TestCases.Login@39ba5a14]" name="Case4" duration-ms="1473" started-at="2018-10-30T18:02:44Z" finished-at="2018-10-30T18:02:45Z">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[msg =>Expected 【密码不能为空】 but found 【null】]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: msg =>Expected 【密码不能为空】 but found 【null】
	at org.testng.Assert.fail(Assert.java:94)
	at com.jmoney.luckeylink.handler.Login_Handler.checkEquals(Login_Handler.java:312)
	at com.jmoney.luckeylink.handler.Login_Handler.Login(Login_Handler.java:288)
	at TestCases.Login.Case4(Login.java:33)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:86)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
	at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
            <line>
              <![CDATA[用例ID: 4]]>
            </line>
            <line>
              <![CDATA[用例名称: 使用正确的账号，空值的密码，无法正常登录成功，且接口返回的实际结果都正确]]>
            </line>
            <line>
              <![CDATA[请求地址: http://120.79.232.23:8000/api/user/login]]>
            </line>
            <line>
              <![CDATA[请求方式: POST]]>
            </line>
            <line>
              <![CDATA[请求参数: username=admin&password=]]>
            </line>
            <line>
              <![CDATA[返回结果: {"password":["该字段不能为空。"],"code":400}]]>
            </line>
          </reporter-output>
        </test-method> <!-- Case4 -->
        <test-method status="PASS" signature="Case5()[pri:0, instance:TestCases.Login@39ba5a14]" name="Case5" duration-ms="1491" started-at="2018-10-30T18:02:45Z" finished-at="2018-10-30T18:02:47Z">
          <reporter-output>
            <line>
              <![CDATA[用例ID: 5]]>
            </line>
            <line>
              <![CDATA[用例名称: 使用错误的账号，错误的密码，无法正常登录成功，且接口返回的实际结果都正确]]>
            </line>
            <line>
              <![CDATA[请求地址: http://120.79.232.23:8000/api/user/login]]>
            </line>
            <line>
              <![CDATA[请求方式: POST]]>
            </line>
            <line>
              <![CDATA[请求参数: username=123456&password=123456789]]>
            </line>
            <line>
              <![CDATA[返回结果: {"code":"999984","msg":"账号或密码错误"}]]>
            </line>
            <line>
              <![CDATA[用例结果: 〖msg〗=>Expected: 【账号或密码错误】, Actual: 【账号或密码错误】]]>
            </line>
          </reporter-output>
        </test-method> <!-- Case5 -->
        <test-method status="PASS" signature="Case6()[pri:0, instance:TestCases.Login@39ba5a14]" name="Case6" duration-ms="1549" started-at="2018-10-30T18:02:47Z" finished-at="2018-10-30T18:02:48Z">
          <reporter-output>
            <line>
              <![CDATA[用例ID: 6]]>
            </line>
            <line>
              <![CDATA[用例名称: 使用错误的账号，正确的密码，无法正常登录成功，且接口返回的实际结果都正确]]>
            </line>
            <line>
              <![CDATA[请求地址: http://120.79.232.23:8000/api/user/login]]>
            </line>
            <line>
              <![CDATA[请求方式: POST]]>
            </line>
            <line>
              <![CDATA[请求参数: username=123456&password=admin369874125]]>
            </line>
            <line>
              <![CDATA[返回结果: {"code":"999984","msg":"账号或密码错误"}]]>
            </line>
            <line>
              <![CDATA[用例结果: 〖msg〗=>Expected: 【账号或密码错误】, Actual: 【账号或密码错误】]]>
            </line>
          </reporter-output>
        </test-method> <!-- Case6 -->
        <test-method status="PASS" signature="Case7()[pri:0, instance:TestCases.Login@39ba5a14]" name="Case7" duration-ms="1499" started-at="2018-10-30T18:02:48Z" finished-at="2018-10-30T18:02:50Z">
          <reporter-output>
            <line>
              <![CDATA[用例ID: 7]]>
            </line>
            <line>
              <![CDATA[用例名称: 使用正确的账号，错误的密码，无法正常登录成功，且接口返回的实际结果都正确]]>
            </line>
            <line>
              <![CDATA[请求地址: http://120.79.232.23:8000/api/user/login]]>
            </line>
            <line>
              <![CDATA[请求方式: POST]]>
            </line>
            <line>
              <![CDATA[请求参数: username=admin&password=123456789]]>
            </line>
            <line>
              <![CDATA[返回结果: {"code":"999984","msg":"账号或密码错误"}]]>
            </line>
            <line>
              <![CDATA[用例结果: 〖msg〗=>Expected: 【账号或密码错误】, Actual: 【账号或密码错误】]]>
            </line>
          </reporter-output>
        </test-method> <!-- Case7 -->
        <test-method status="PASS" signature="TearDown()[pri:0, instance:TestCases.Login@39ba5a14]" name="TearDown" is-config="true" duration-ms="1854" started-at="2018-10-30T18:02:50Z" finished-at="2018-10-30T18:02:52Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- TearDown -->
      </class> <!-- TestCases.Login -->
    </test> <!-- 用户登录接口 -->
  </suite> <!-- Web后端-接口自动化测试 -->
</testng-results>
